### PR TITLE
Migrate to Font-Awesome 5

### PR DESCRIPTION
**Pull request by stevenjoezhang**
Issue resolved: https://github.com/theme-next/hexo-theme-next/issues/505

### CODE REVIEW

File: layout/_macro/post.swig [hunk 1]
```diff
@@ -49,7 +49,7 @@
           {%- if theme.post_meta.created_at %}
             <span class="post-meta-item">
               <span class="post-meta-item-icon">
-                <i class="fa fa-calendar-o"></i>
+                <i class="far fa-calendar"></i>
```
Comment (sli1989) [inline layout/_macro/post.swig]: `far`?

File: layout/_partials/post/post-copyright.swig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{%- set ccIcon = '<i class="fa fa-fw fa-creative-commons"></i>' %}
+{%- set ccIcon = '<i class="fab fa-fw fa-creative-commons"></i>' %}
```
Comment (sli1989) [inline layout/_partials/post/post-copyright.swig]: just fused about `fa`/`far`/`fab`? a huge update. thanks.

File: layout/_macro/post.swig (hunk 1 above)
Comment (jiangtj) [reply layout/_macro/post.swig]: Some `far` are free, and some of the original v4 icons are `far` in v5

File: layout/_partials/post/post-copyright.swig (hunk 2 above)
Comment (stevenjoezhang) [reply layout/_partials/post/post-copyright.swig]: I did not find the official documentation for migration from v4.7 to v5. In fact, this PR is based on this gist: https://github.com/estebandelaf/scripts/blob/master/fontawesome4to5.sh

About new class names:
`fas` for `solid`
`far` for `regular`
`fab` for `brand`

**Comment by stevenjoezhang** [comment]
Shall we merge this PR to v7.8.0 release (March 1st)?

**Pull request merged**